src/snowflake/snowpark/session.py

```diff
@@ -3456,6 +3459,79 @@ def flatten(
         set_api_call_source(df, "Session.flatten")
         return df
 
+    def register_profiler(self, profiler: Profiler):
+        """Register a profiler to a session, all action are actually executed during this function"""
+        if (
+            profiler.session is not None
+            and profiler.session._session_id != self._session_id
+        ):
+            raise ValueError("A profiler can only be registered to one session.")
+        self.profiler = profiler
+        self.profiler.session = self
+        if (
+            len(self.sql(f"show stages like '{profiler.stage}'").collect()) == 0
+            and len(
+                self.sql(
+                    f"show stages like '{profiler.stage.split('.')[-1]}'"
+                ).collect()
+            )
+            == 0
+        ):
+            self.sql(
+                f"create temp stage if not exists {profiler.stage} FILE_FORMAT = (RECORD_DELIMITER = NONE FIELD_DELIMITER = NONE )"
+            ).collect()
+        self.profiler._register_modules()
+        self.profiler._set_targeted_stage()
+        self.profiler._set_active_profiler()
+        self.profiler.query_history = self.query_history()
+
+    def show_profiles(self) -> str:
+        """
+        Return and show the profiles of last executed stored procedure.
+
+        Note:
+            This function must be called right after the execution of stored procedure you want to profile.
+        """
+        if self.profiler is not None and isinstance(self.profiler, Profiler):
+            return self.profiler.show_profiles()
+        else:
+            raise ValueError(
+                "profiler is not set, use session.register_profiler or profiler context manager"
+            )
+
+    def dump_profiles(self, dst_file: str):
+        """
+        Write the profiles of last executed stored procedure to given file.
+
+        Note:
+            This function must be called right after the execution of stored procedure you want to profile.
+
+        Args:
+            dst_file: String of file name that you want to store the profiles.
+        """
+        if self.profiler is not None and isinstance(self.profiler, Profiler):
+            self.profiler.dump_profiles(dst_file=dst_file)
+        else:
+            raise ValueError(
+                "profiler is not set, use session.register_profiler or profiler context manager"
+            )
+
+    def register_profiler_modules(self, stored_procedures: List[str]):
+        """
+        Register stored procedures to generate profiles for them.
+
+        Note:
+            Registered nodules will be overwritten by this function,
```

Comment:
nodules -> modules

Reply:
Problem: Comma splice.
Resolution: Change comma to period and capitalize the first word of the second sentence.

In other words, change it to this:
Registered modules will be overwritten by this function. Using this function with an empty string will remove registered modules.